src/Hatchinq/Util.elm

```diff
@@ -40,27 +40,60 @@ keysDownAttribute keyCodes =
             in
             case maybeMessage of
                 Just message ->
-                    Json.Decode.succeed message
+                    Decode.succeed message
 
                 Nothing ->
-                    Json.Decode.fail "no keyCode found"
+                    Decode.fail "no keyCode found"
     in
-    Element.htmlAttribute <| Html.Events.on "keydown" (Json.Decode.andThen isKey Html.Events.keyCode)
+    Element.htmlAttribute <| Html.Events.on "keydown" (Decode.andThen isKey Html.Events.keyCode)
 
 
 keyDownAttribute : Int -> msg -> Element.Attribute msg
 keyDownAttribute keyCode message =
     let
         isKey code =
             if code == keyCode then
-                Json.Decode.succeed message
+                Decode.succeed message
 
             else
-                Json.Decode.fail ("not keyCode " ++ String.fromInt keyCode)
+                Decode.fail ("not keyCode " ++ String.fromInt keyCode)
     in
-    Element.htmlAttribute <| Html.Events.on "keydown" (Json.Decode.andThen isKey Html.Events.keyCode)
+    Element.htmlAttribute <| Html.Events.on "keydown" (Decode.andThen isKey Html.Events.keyCode)
 
 
 takeFirstNLines : String -> Int -> String
 takeFirstNLines text numLines =
     String.join "\n" (List.take numLines (String.split "\n" text))
+
+
+onClickPropagation : Bool -> msg -> Element.Attribute msg
```

Comment:
`onClickPropagation True` reads a bit confusing. I would simply drop the parameter and name it explicitly `onClickWithoutPropagation` or something similar. This function we anyways use only like that. For all other cases, there is the plain `onClick` event.

Reply:
Then I would still name it `onClickWithoutPropagation True`. sounds a bit more intuitive I would say :)